### PR TITLE
helm upgrade check logic

### PR DESCRIPTION
**Pull request by JesseGoodier**
## What does this PR change?

1. fix boolean logic on v2 upgrade with multi-cluster
2. remove block on amp/mimir/gmp v2 upgrades

## Does this PR rely on any other PRs?
No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)
Added support in Kubecost 2.1 for multi-cluster AMP/GMP/Mimir 

## What risks are associated with merging this PR? What is required to fully test this PR?
Unknown/untested issues with one of the platforms.

## How was this PR tested?
ADOT QA environment.

## Have you made an update to documentation? If so, please provide the corresponding PR.
todo

### CODE REVIEW

File: cost-analyzer/crds/cluster-turndown-crd.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if (.Values.clusterController).enabled }}
```
Comment (jessegoodier) [inline cost-analyzer/crds/cluster-turndown-crd.yaml]: @ameijer 
does this make sense? Is there a use case for the CRD if the controller isn't enabled?

Comment (ameijer) [reply cost-analyzer/crds/cluster-turndown-crd.yaml]: yes, however, the CRDs directory doesn't support templating :(

**Comment by jessegoodier** [comment]
closing, found new bug

**Pull request closed without merging**